api/data_test.go

```diff
@@ -191,6 +191,44 @@ func TestCreateNewCacheTime(t *testing.T) {
 		})
 	})
 }
+
+func TestGetCacheTimeReturnsError400(t *testing.T) {
+	Convey("Given an api", t, func() {
+		ctx := context.Background()
+		dataStoreMock := &mock.DataStoreMock{}
+		dataStoreAPI := setupAPIWithStore(ctx, dataStoreMock)
+		Convey("When the id provided is not 32 characters in length", func() {
```

Comment:
Also small - but as we are testing the same isValidID function inside `CreateOrUpdateCacheTimeReturnsErr`, these scenarios should be covered. Perhaps refactoring this test into one test where we check`responseRecorder.Body.String()` contains all three ID validation errors (!ID length 32, !lowercase, !hexadecimal) would reduce the repeated code. Please let me know if this does not make sense.

Reply:
I wouldn't personally do this, because what makes for good production code does not necessarily make for good testing code. It's OK to have some repetition in the tests, because it's more important to have clear intentions and easy to read tests.

By applying your suggestion, we wouldn't reduce that much repetition and the resulting tests would be less expressive, I think.

Reply:
That makes sense, also happy with that 😄

Reply:
this has been discussed and we will tackle this in the refactoring ticket for tests